### PR TITLE
Refactoring of coordinate changes in time-propagation

### PR DESCRIPTION
**Pull request by vanderhe**
Moves coordinate updates and sparse-array re-initialization from `buildH0S()` to dedicated `handleCoordinateChange()` routine + a few unrelated changes (sorry).

### CODE REVIEW

**Review by bhourahine — APPROVED** [review]
Looks good, might want to think about comment on cutoffs.

File: src/dftbp/type/cutoff.F90 [hunk 1]
```diff
@@ -0,0 +1,41 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2024  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+!> Contains a type with cutoff information.
```
[the remark this answers is not in the document]
Comment (bhourahine) [reply src/dftbp/type/cutoff.F90:8]: Depends, if the module will grow to contain the majority of cutoff information (SCC, repulsives, dispersion, ...) in some form then it's name would be a good match. Similarly if the extension will be within this type or if a second type gets added to the module... On the other hand if this is the planned extent of the module and type, then the name for both the structure and module is somewhat too general IMO.

[the remark this answers is not in the document]
Comment (vanderhe) [reply src/dftbp/type/cutoff.F90]: @bhourahine I've renamed the file/module to `eleccutoffs.F90` and slightly adapted the module's description. Does this work for you?

File: src/dftbp/type/eleccutoffs.F90 [hunk 2]
```diff
@@ -0,0 +1,41 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2024  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+!> Contains types with electronic cutoff information.
```
Comment (bhourahine) [inline src/dftbp/type/eleccutoffs.F90:8]:
```suggestion
!> Contains types with electronic/hamiltonian interaction cutoff information.
```

**Review by bhourahine — APPROVED** [review]
LGTM

**Comment by bhourahine** [comment]
@vanderhe Might be worth putting it through buildbot just in case.

File: src/dftbp/type/eleccutoffs.F90 (hunk 2 above)
Comment (vanderhe) [reply src/dftbp/type/eleccutoffs.F90]: Done.

**Comment by bhourahine** [comment]
> > @vanderhe Might be worth putting it through buildbot just in case.
> 
> Sure, did you already trigger the tests?

No, do you want to?

**Comment by vanderhe** [comment]
> > > @vanderhe Might be worth putting it through buildbot just in case.
> > 
> > 
> > Sure, did you already trigger the tests?
> 
> No, do you want to?

Will do.

**Comment by vanderhe** [comment]
@bhourahine I see two socket tests failing, everything else passes.

**Comment by vanderhe** [comment]
@bhourahine Passed all tests, with the MPI zombie processes killed.

**Pull request merged**